core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -319,6 +319,39 @@ impl EthNamespace {
         Ok(tx_count?.map(|(_, count)| count))
     }
 
+    pub async fn get_block_receipts_impl(
+        &self,
+        block_id: BlockId,
+    ) -> Result<Vec<TransactionReceipt>, Web3Error> {
+        const METHOD_NAME: &str = "get_block_receipts";
+
+        let method_latency = API_METRICS.start_block_call(METHOD_NAME, block_id);
+
+        let transaction_count = self
+            .get_block_transaction_count_impl(block_id)
+            .await?
+            .ok_or(Web3Error::NoBlock)?;
+
+        let mut receipts = Vec::with_capacity(transaction_count.as_usize());
+
+        for index in 0..transaction_count.as_usize() {
+            let transaction = self
+                .get_transaction_impl(TransactionId::Block(block_id, index.into()))
```

Comment:
This doesn't look efficient. I'd at least add a method to DAL to get all transaction hashes for `block_id` (maybe using `BlocksWeb3Dal::get_block_by_web3_block_id()`, although it loads some unnecessary data), + transactions with the given list of hashes, slightly generalizing `TransactionsWeb3Dal::get_transaction_receipt()`). In this case, we'd have O(1) DB queries instead of the linear number.

Reply:
You are right, but normally we have 20-30 to 150 txs per block